plugin/register.go

```diff
@@ -0,0 +1,57 @@
+package plugin
+
+import (
+	"fmt"
+	"github.com/mitchellh/cli"
+	"reflect"
+	"strings"
+	"trellis-cli/cmd"
+)
+
+func Register(c *cli.CLI, searchPaths []string, validPluginFilenamePrefixes []string) {
+	coreNamespaces := namespacesFor(reflect.ValueOf(c.Commands))
+	pluginFinder := newFinder(validPluginFilenamePrefixes, searchPaths, coreNamespaces)
+	plugins := pluginFinder.find()
+	pluginNamespaces := namespacesFor(reflect.ValueOf(plugins))
+
+	// Register plugin commands.
+	for name, bin := range plugins {
+		pluginCommand := cmd.NewPassthroughCommand(name, bin, c.Args)
+		c.Commands[name] = func() (cli.Command, error) {
+			return pluginCommand, nil
+		}
+	}
+
+	// Separate plugin commands from core command lists
+	c.HiddenCommands = append(c.HiddenCommands, pluginNamespaces...)
+	// Append plugin command list to help text.
+	c.HelpFunc = helpFunc(c.Name, pluginNamespaces)
+}
+
+func namespacesFor(v reflect.Value) []string {
+	m := v.MapKeys()
+
+	var namespaces []string
+	for _, v := range m {
+		name := fmt.Sprint(v)
+		namespace := strings.Split(name, " ")[0]
```

Comment:
```go
		namespace := strings.Split(v.String(), " ")[0]

```

`fmt.Sprint` is just calling `name.String()` under the hood so think it's better to call it directly.

Reply:
done in [`44045b0` (#144)](https://github.com/roots/trellis-cli/pull/144/commits/44045b0d0fdbe969b57270296c824a39264b2d26)